From: jbeulich@suse.com
Subject: netback: fix flipping mode
Patch-mainline: Never, SUSE-Xen specific
References: bsc#996664

Filling the grant transfer operation's MFN value needs to be done
before changing the PFN <-> MFN translation. Move the setting of all
gop fields ahead.

--- a/drivers/xen/netback/netback.c
+++ b/drivers/xen/netback/netback.c
@@ -416,6 +416,12 @@ static u16 netbk_gop_frag(netif_t *netif
 		copy_gop->len = size;
 	} else {
 		meta->copy = 0;
+
+		gop = npo->trans + npo->trans_prod++;
+		gop->mfn = virt_to_mfn(page_address(page));
+		gop->domid = netif->domid;
+		gop->ref = req->gref;
+
 		if (!xen_feature(XENFEAT_auto_translated_physmap)) {
 			unsigned long new_mfn = alloc_mfn(netbk);
 
@@ -437,11 +443,6 @@ static u16 netbk_gop_frag(netif_t *netif
 				MMU_MACHPHYS_UPDATE;
 			mmu->val = page_to_pfn(page);
 		}
-
-		gop = npo->trans + npo->trans_prod++;
-		gop->mfn = virt_to_mfn(page_address(page));
-		gop->domid = netif->domid;
-		gop->ref = req->gref;
 	}
 	return req->id;
 }
